lib/realtime/api.ex

```diff
@@ -243,4 +256,24 @@ defmodule Realtime.Api do
   end
 
   defp maybe_restart_db_connection(_changeset), do: nil
+
+  defp local_call? do
+    region = Application.get_env(:realtime, :region)
+    master_region = Application.get_env(:realtime, :master_region) || region
+    region == master_region
+  end
+
+  defp region_aware_write(%struct{} = argument, operation, tenant_id) when struct in [Changeset, Ecto.Query] do
+    if local_call?(),
+      do: local_call(operation, [argument], tenant_id),
+      else: remote_call(operation, [argument], tenant_id)
+  end
+
+  defp local_call(operation, args, _tenant_id), do: apply(Realtime.Repo, operation, args)
+
+  defp remote_call(operation, args, tenant_id) do
+    master_region = Application.get_env(:realtime, :master_region)
+    {:ok, master_node} = Nodes.node_from_region(master_region, self())
```

Comment:
Do we want to treat the error case and log something and return a `{:error, _}` ? We might want a test for this as well. 

The functions that end up calling this like `create_tenant` expects {:ok, tenant} or `{:error, changeset}` I think?

We can write a unit test for this case by stubbing `Nodes.nodes_from_region` to return `{:error, :not_available}`